model/database/cosmosdb-metrics.yaml

```diff
@@ -26,3 +26,68 @@ groups:
       - ref: server.port
         requirement_level:
           conditionally_required: If using a port other than the default port for this DBMS and if `server.address` is set.
+
+  - id: metric.db.client.cosmosdb.request.duration
+    type: metric
+    metric_name: db.client.cosmosdb.request.duration
+    brief: "End to end Latency of Azure Cosmos DB Backend Service network call"
+    instrument: histogram
+    unit: "s"
+    stability: experimental
+    extends: attributes.db.cosmosdb.network.minimal
+
+  - id: metric.db.client.cosmosdb.request.body.size
+    type: metric
+    metric_name: db.client.cosmosdb.request.body.size
+    brief: "Size of request object sent to Azure Cosmos DB backend"
+    instrument: histogram
+    unit: "{bytes}"
+    stability: experimental
+    extends: attributes.db.cosmosdb.network.minimal
+
+  - id: metric.db.client.cosmosdb.response.body.size
+    type: metric
+    metric_name: db.client.cosmosdb.response.body.size
+    brief: "Size of response received from Azure Cosmos DB backend"
+    instrument: histogram
+    unit: "{bytes}"
+    stability: experimental
+    extends: attributes.db.cosmosdb.network.minimal
+
+  - id: metric.db.client.cosmosdb.request.channel_aquisition.duration
+    type: metric
+    metric_name: db.client.cosmosdb.request.channel_aquisition.duration
+    brief: "Time taken in creating connection, It is only applicable in Direct Mode."
+    instrument: histogram
+    unit: "s"
+    stability: experimental
+    extends: attributes.db.cosmosdb.network.minimal
+
+  - id: metric.db.server.cosmosdb.request.duration
+    type: metric
+    metric_name: db.server.cosmosdb.request.duration
+    brief: "Backend Server Latency, i.e. time taken by Azure Cosmos DB server to process the request. It is only applicable in Direct Mode."
+    instrument: histogram
+    unit: "s"
+    stability: experimental
+    extends: attributes.db.cosmosdb.network.minimal
+
+  - id: metric.db.client.cosmosdb.request.transit.duration
+    type: metric
+    metric_name: db.client.cosmosdb.request.transit.duration
+    brief: >
+      "Transit time refers to the duration taken by a request to travel across the network—from the moment it leaves the client to
+      when the response is received back at the client. It is only applicable in Direct Mode."
+    instrument: histogram
+    unit: "s"
+    stability: experimental
+    extends: attributes.db.cosmosdb.network.minimal
+
+  - id: metric.db.client.cosmosdb.request.received.duration
```

Comment:
you can get time-in-SDK by subtracting `db.client.operation.duration` from `metric.db.client.cosmosdb.request.duration`. Why is this metric necessary? 

Also, have you tried implementing it? It'd be really hard to know when request is "fully processed by the SDK" given layers of convenience APIs, paging and other things.

Reply:
https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/troubleshoot-dotnet-sdk-slow-request?tabs=cpu-new#requesttimeline we already have this information for `rntbd` calls and for we are not going to capture it for `HTTP` calls